packages/babel-core/src/config/helpers/config-api.ts

```diff
@@ -126,7 +126,10 @@ function assertVersion(range: string | number): void {
     throw new Error("Expected string or integer value.");
   }
 
-  if (semver.satisfies(coreVersion, range)) return;
+  // We want "*" to also allow any pre-release, but we do not pass
+  // the includePrerelease option to semver.satisfies because we
+  // do not want ^7.0.0 to match 8.0.0-alpha.1.
+  if (range === "*" || semver.satisfies(coreVersion, range)) return;
```

Comment:
This is needed when running tests locally, however until when we release this PR in Babel 8 we need to manually apply the changes to `cjs-proxy.cjs`